src/compiler/scala/tools/nsc/backend/jvm/analysis/BackendUtils.scala

```diff
@@ -28,17 +28,17 @@ import scala.util.control.{NoStackTrace, NonFatal}
  *
  * TODO: move out of `analysis` package?
  */
-abstract class BackendUtils extends PerRunLazy {
+abstract class BackendUtils extends PerRunInit {
   val postProcessor: PostProcessor
 
-  import postProcessor.{bTypes, bTypesFromClassfile, callGraph, frontendAccess}
+  import postProcessor.{bTypes, bTypesFromClassfile, callGraph}
   import bTypes._
   import callGraph.ClosureInstantiation
   import coreBTypes._
   import frontendAccess.compilerSettings
 
   // unused objects created by these constructors are eliminated by pushPop
-  private[this] lazy val sideEffectFreeConstructors: LazyVar[Set[(String, String)]] = perRunLazy {
+  private[this] lazy val sideEffectFreeConstructors: LazyVar[Set[(String, String)]] = perRunLazy(this) {
```

Comment:
Additonally, the blend of locks based on lazy vals and `frontEndLock` needs to be considered carefully for potential deadlocks.